carbon-design-system/carbon
# fix: ComboBox readonly state fixed

Closes #17382 

Fixed the Combobox's Read Only state, which previously would allow the user to interact with the list box.

This modification has been done after referring the read only states of Dropdown and Multiselect.
Dropdown's readonly story for reference:
https://react.carbondesignsystem.com/?path=/story/components-dropdown--playground&args=readOnly:!true

#### Changelog

**New**

- Added read only prop check to stop the list box opening.
- Added onFocus and onClick event handlers for readOnlyEventHandlers.
- Added `event.preventDefault();` for Tab interaction for readOnly prop



#### Testing / Reviewing

- Navigate to the playground story of Combobox component
- Turn on the readOnly prop
- check the interaction and make sure it doesn't open or allows text focus, only the box focus is present.
